Client/Frontend/Browser/TabTrayController.swift

```diff
@@ -5,24 +5,214 @@
+        self.backgroundHolder.layer.shadowColor = UIColor.blackColor().CGColor
+        self.backgroundHolder.layer.shadowOffset = CGSizeMake(0,0)
+        self.backgroundHolder.layer.shadowOpacity = 0.25
+        self.backgroundHolder.layer.shadowRadius = 2.0
+
+        self.background = UIImageViewAligned()
+        self.background.contentMode = UIViewContentMode.ScaleAspectFill
+        self.background.clipsToBounds = true
+        self.background.userInteractionEnabled = false
+        self.background.alignLeft = true
+        self.background.alignTop = true
+
+        self.title = UITextView()
+        self.title.backgroundColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.75)
+        self.title.textColor = UIColor.whiteColor()
+        self.title.textAlignment = NSTextAlignment.Left
+        self.title.userInteractionEnabled = false
+
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        backgroundHolder.addSubview(self.background)
+        addSubview(backgroundHolder)
+        addSubview(self.title)
+        backgroundColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0)
+
+        self.title.addObserver(self, forKeyPath: "contentSize", options: .New, context: nil)
+    }
+
+    deinit {
+        self.title.removeObserver(self, forKeyPath: "contentSize")
+    }
+
+    private override func observeValueForKeyPath(keyPath: String, ofObject object: AnyObject, change: [NSObject : AnyObject], context: UnsafeMutablePointer<Void>) {
+        let tv = object as UITextView
+        verticalCenter(tv)
+    }
+
+    private func verticalCenter(text: UITextView) {
+        var  top = (TextBoxHeight - text.contentSize.height) * text.zoomScale / 2.0
+        top = top < 0.0 ? 0.0 : top
+        text.contentOffset = CGPoint(x: 0, y: -top)
+    }
+
+    required init(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func showFullscreen(container: UIView, table: UITableView) {
+        margin = 0
+        container.insertSubview(self, atIndex: container.subviews.count)
+        frame = CGRect(x: container.frame.origin.x,
+            y: container.frame.origin.y + ToolbarHeight + StatusBarHeight,
+            width: container.frame.width,
+            height: container.frame.height)
+        title.alpha = 0
+        setNeedsLayout()
+    }
+
+    func showAt(offsetY: Int, container: UIView, table: UITableView) {
+        margin = Margin
+        container.insertSubview(self, atIndex: container.subviews.count)
+        frame = CGRect(x: 0,
+            y: ToolbarHeight + StatusBarHeight + CGFloat(offsetY) * CellHeight - table.contentOffset.y,
+            width: container.frame.width,
+            height: CellHeight)
+        title.alpha = 1
+        setNeedsLayout()
+    }
+
+    var tab: Browser? {
+        didSet {
+            if let tab = self.tab {
+                background.image = tab.screenshot()
+                title.text = tab.webView.title
```

Comment:
Can you create a title property in browser that forwards the webView title like we do with URL? I've been trying to avoid exposing webView wherever possible. It would be nice if we had a way to make webView exposed only to BVC instead of manually enforcing this...